packages/hardhat-verify/src/index.ts

```diff
@@ -211,9 +216,42 @@ subtask(TASK_VERIFY_RESOLVE_ARGUMENTS)
 /**
  * Returns a list of verification subtasks.
  */
-subtask(TASK_VERIFY_GET_VERIFICATION_SUBTASKS, async (): Promise<string[]> => {
-  return [TASK_VERIFY_ETHERSCAN];
-});
+subtask(
+  TASK_VERIFY_GET_VERIFICATION_SUBTASKS,
+  async (_, { config }): Promise<string[]> => {
+    const verificationSubtasks = [];
+
+    if (config.etherscan.enabled) {
+      verificationSubtasks.push(TASK_VERIFY_ETHERSCAN);
+    }
+
+    if (config.sourcify.enabled) {
+      verificationSubtasks.push(TASK_VERIFY_SOURCIFY);
+    } else {
```

Comment:
Shouldn't this be something like:

```js
} else if (userConfig.sourcify?.enabled !== undefined) {
```

Reply:
The `config.sourcify` key is initialized in the config extender. I don't think we need to use `userConfig` after that. Am I missing your point?

Reply:
Self note: this is to distinguish between the default (`false` value, we show the warning) and the explicit `enabled: false` in the config (we do not show the warning).